pkg/migrations/mls/20240411200242_init-identity.up.sql

```diff
@@ -0,0 +1,34 @@
+SET statement_timeout = 0;
+
+--bun:split
+
+CREATE TABLE inbox_log (
+    sequence_id BIGSERIAL PRIMARY KEY,
+    inbox_id TEXT NOT NULL,
+    commit_status SMALLINT NOT NULL,
+    server_timestamp_ns BIGINT NOT NULL,
+    identity_update_proto BYTEA NOT NULL
+);
+
+--bun:split
+
+CREATE INDEX idx_inbox_log_inbox_id_commit_status ON inbox_log(inbox_id, commit_status);
+
+--bun:split
+
+CREATE TABLE address_log (
+    inbox_log_sequence_id BIGINT PRIMARY KEY,
+    address TEXT NOT NULL,
+    identity_update_proto BYTEA NOT NULL
+);
+
+--bun:split
+
+CREATE INDEX idx_address_log_address ON address_log(address);
+
+--bun:split
+
+CREATE TABLE address_lookup_cache (
```

Comment:
I wonder if we can merge the `address_lookup_cache` and the `address_log` into one thing.

What if we had a table like this:

```sql
CREATE TABLE address_history (
     address TEXT NOT NULL,
     inbox_id TEXT NOT NULL,
     inbox_log_sequence_id BIGINT NOT NULL,
     
     PRIMARY KEY (address, inbox_log_sequence_id)
)
```

This would be an append-only table, where we would write new entries on each valid update that touches an address. It gives us a history of updates, and makes it easy to query for the current update

To read the current value we would do something like 
```sql
SELECT DISTINCT ON (address) inbox_id WHERE address IN (?) ORDER BY inbox_log_sequence_id DESC;
```

With the right index on address/inbox_log_sequence_id it should still be very fast to read.

Reply:
Or maybe a GROUP BY, which is usually faster. But the same idea

Reply:
We'd have to benchmark and see if it makes the wrong trade-off in terms of read vs write performance. But if it's using the indexes correctly, it might be roughly equivalent.